### PR TITLE
Setup NS cache in systemd-resolved

### PR DESCRIPTION
**Pull request by aegiap**
New version of Linux distribution ship systemd-resolved. This PR add the associated nameserver cache for the datacenter of the VM to systemd-resolved configuration.

Validated on `Ubuntu 18.04 bionic`.

Signed-off-by: aegiap <aegiap@gandi.net>

### CODE REVIEW

File: swap_init.py [hunk 1]
```diff
@@ -192,16 +198,28 @@ def resolver_setup(nameservers):
...
         rlink = '%s/tail' % rconfdir
         if os.path.exists(rlink):
             os.unlink(rlink)
         os.symlink('%s/original' % rconfdir, rlink)
+    # systemd-resolved
+    if os.path.exists('/etc/systemd/resolved.conf'):
+        dns_list=resolver_gen(nameservers, 'dnslist')
+        rpath='/etc/systemd/resolved.conf.d'
+        if not os.path.exists(rpath):
+            os.mkdir(rpath, 0755)
```
Comment (David-5-1) [inline swap_init.py]: If you want to be compatible with python 3, you can use `0o755` for octal numbers.

Comment (aegiap) [reply swap_init.py]: @David-5-1 thanks, fixed now.

File: swap_init.py [hunk 2]
```diff
@@ -181,27 +184,44 @@ def resolver_gen(nameservers):
...
+    # systemd-resolved
+    if os.path.exists('/etc/systemd/resolved.conf'):
+        dns_list=resolver_gen(nameservers, 'dnslist')
+        rpath='/etc/systemd/resolved.conf.d'
+        if not os.path.exists(rpath):
+            os.mkdir(rpath, 0o755)
+        with open('%s/gandi.conf' % rpath, 'w') as f:
+            f.write('[Resolve]\nDNS=%s\n' % dns_list)
+        # XXX we may have to move this to systemd-networkd in the futur
+        subprocess.Popen(['/bin/ip', '-4', 'addr', 'add', '127.0.0.53/32', 'dev', 'lo']).wait()
```
Comment (David-5-1) [inline swap_init.py]: A manual test creating `/etc/systemd/resolved.conf.d/gandi.conf` followed by a restart of `systemd-resolved` seems to indicate that this is not needed.

Comment (aegiap) [reply swap_init.py]: Indeed, after another test on my setup, I removed the ip addr command and the two lines in /etc/network/interfaces.

**Comment by aegiap** [comment]
This PR also change the default network interfaces Debian configuration to source the files in /etc/network/interfaces.d/*

**Pull request merged**